src/Core/Generator/SchemaGenerator.cs

```diff
@@ -44,7 +44,7 @@ private SchemaGenerator(List<JsonDocument> data, string containerName, RuntimeCo
                 // Populate entity and singular name mapping if configuration is provided.
                 foreach (KeyValuePair<string, Entity> item in config.Entities)
                 {
-                    _entityAndSingularNameMapping.Add(item.Value.GraphQL.Singular, item.Key);
+                    _entityAndSingularNameMapping.Add(item.Value.GraphQL.Singular.Pascalize(), item.Key);
```

Comment:
why it needs to be pascalized?

Reply:
found a bug where if singular/ plural are defined in pascalized form(which is valid entry IMO), then it was failing while generating schema.